### ISSUE TITLE
The MSVC11 (Visual Studio 2012) throw std::tuple error when compiling

### ISSUE THREAD

**Issue by LevskiWeng** [cppformat/cppformat#148]
When compiling in MSVC11, the compiler said:
> error C2977: 'std::tuple' : too many template arguments

I've googled the error message, and according to the [stackoverflow post](http://stackoverflow.com/questions/8274588/c2977-stdtuple-too-many-template-arguments-msvc11), I added the following preprocessor to all cppformat projects:
> _VARIADIC_MAX=10

and then it compiled successfully.

I wonder is it a good idea for the project maintainers add the above preprocessor to the default `CMakefile.txt` when CMake found that the build environment is MSVC11?

**Comment by vitaut** [comment]
Thanks for the bug report. I don't have access to MSVC11 at the moment, but I think https://github.com/cppformat/cppformat/commit/f6823221d32323825c1f317c1d7dc44068743b00 should fix the problem. Could you please check if it does?

**Comment by LevskiWeng** [comment]
Thanks for your quick patch. But I'm afraid your patch doesn't work for me.

I replaced your modified `CMakeLists.txt` and tested the following `cmake` command:
> cmake "C:\Absolute path of cppformat source directory"

But I failed to compile the solution again with the same compilation error.

I roughly glanced at the build-in implementation of `std::tuple<>` in MSVC11, it uses **macro** to expand the number to template parameters (MSVC11 seems not able to fully support variadic templates). The default max supported number of template parameters in my environment is 8, so the compiler complained ` 'std::tuple' : too many template arguments` only when the number of template parameter exceeded 8 (Line 9741 in `cppformat/gmock/gtest/gtest.h`)

Any better ideas?
